pkg/apis/v1/nodeclaim.go

```diff
@@ -0,0 +1,189 @@
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// NodeClaimSpec describes the desired state of the NodeClaim
+type NodeClaimSpec struct {
+	// Taints will be applied to the NodeClaim's node.
+	// +optional
+	Taints []v1.Taint `json:"taints,omitempty"`
+	// StartupTaints are taints that are applied to nodes upon startup which are expected to be removed automatically
+	// within a short period of time, typically by a DaemonSet that tolerates the taint. These are commonly used by
+	// daemonsets to allow initialization and enforce startup ordering.  StartupTaints are ignored for provisioning
+	// purposes in that pods are not required to tolerate a StartupTaint in order to have nodes provisioned for them.
+	// +optional
+	StartupTaints []v1.Taint `json:"startupTaints,omitempty"`
+	// Requirements are layered with GetLabels and applied to every node.
+	// +kubebuilder:validation:XValidation:message="requirements with operator 'In' must have a value defined",rule="self.all(x, x.operator == 'In' ? x.values.size() != 0 : true)"
+	// +kubebuilder:validation:XValidation:message="requirements operator 'Gt' or 'Lt' must have a single positive integer value",rule="self.all(x, (x.operator == 'Gt' || x.operator == 'Lt') ? (x.values.size() == 1 && int(x.values[0]) >= 0) : true)"
+	// +kubebuilder:validation:XValidation:message="requirements with 'minValues' must have at least that many values specified in the 'values' field",rule="self.all(x, (x.operator == 'In' && has(x.minValues)) ? x.values.size() >= x.minValues : true)"
+	// +kubebuilder:validation:MaxItems:=30
+	// +required
+	Requirements []NodeSelectorRequirementWithMinValues `json:"requirements" hash:"ignore"`
+	// Resources models the resource requirements for the NodeClaim to launch
+	// +optional
+	Resources ResourceRequirements `json:"resources,omitempty" hash:"ignore"`
+	// Kubelet defines args to be used when configuring kubelet on provisioned nodes.
+	// They are a subset of the upstream types, recognizing not all options may be supported.
+	// Wherever possible, the types and names should reflect the upstream kubelet types.
+	// +kubebuilder:validation:XValidation:message="imageGCHighThresholdPercent must be greater than imageGCLowThresholdPercent",rule="has(self.imageGCHighThresholdPercent) && has(self.imageGCLowThresholdPercent) ?  self.imageGCHighThresholdPercent > self.imageGCLowThresholdPercent  : true"
+	// +kubebuilder:validation:XValidation:message="evictionSoft OwnerKey does not have a matching evictionSoftGracePeriod",rule="has(self.evictionSoft) ? self.evictionSoft.all(e, (e in self.evictionSoftGracePeriod)):true"
+	// +kubebuilder:validation:XValidation:message="evictionSoftGracePeriod OwnerKey does not have a matching evictionSoft",rule="has(self.evictionSoftGracePeriod) ? self.evictionSoftGracePeriod.all(e, (e in self.evictionSoft)):true"
+	Kubelet *KubeletConfiguration `json:"kubelet,omitempty"`
```

Comment:
Kubelet is getting dropped from this config right? Since we are shifting that field over to the EC2NodeClass?

Reply:
I'm planning on removing this field and other addition to the V1 APIs in a separate PR

Reply:
I'm fine with this methodology, can we rename the PR then to indicate that this is just adding a copied version of the old version? As it reads, it sounds like this is introducing the new changes. I'd probably prefer that we just merge what we want to have at v1 from the outset, but merging the duplicate and taking some away after the fact also sounds reasonable, as long as we're clear about it.